app/src/main/java/com/kickstarter/ui/activities/CommentsActivity.kt

```diff
@@ -0,0 +1,195 @@
+            .compose(bindToLifecycle())
+            .subscribe { viewModel.inputs.commentBodyChanged(it) }
+        viewModel.outputs.currentCommentBody()
+            .compose(Transformers.takePairWhen(commentBodyEditText))
+            .observeOn(AndroidSchedulers.mainThread())
+            .compose(bindToLifecycle())
+            .subscribe { it.second?.append(it.first) }
+        viewModel.outputs.commentsData()
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe { adapter?.takeData(it) }
+        viewModel.outputs.enablePostButton()
+            .compose(Transformers.combineLatestPair(postCommentButton))
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe { setPostButtonEnabled(it.second, it.first) }
+        viewModel.outputs.commentButtonHidden()
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(ViewUtils.setGone(binding.commentsToolbar.commentButton))
+        viewModel.outputs.showCommentDialog()
+            .filter { it != null }
+            .map { it.first }
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe { showCommentDialog(it) }
+        alertDialog
+            .compose(Transformers.takeWhen(viewModel.outputs.dismissCommentDialog()))
+            .observeOn(AndroidSchedulers.mainThread())
+            .compose(bindToLifecycle())
+            .subscribe { dismissCommentDialog(it) }
+        lifecycle()
+            .compose(Transformers.combineLatestPair(alertDialog))
+            .filter { it.first == ActivityEvent.DESTROY }
+            .map { it.second }
+            .observeOn(AndroidSchedulers.mainThread())
+            // NB: We dont want to bind to lifecycle because we want the destroy event.
+            // .compose(bindToLifecycle())
+            .take(1)
+            .subscribe { dismissCommentDialog(it) }
+        toastMessages()
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(ViewUtils.showToast(this))
+
+        findViewById<View>(R.id.project_context_view)?.setOnClickListener { projectContextViewClick() }
+        findViewById<View>(R.id.comment_button)?.setOnClickListener { commentButtonClicked() }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        recyclerViewPaginator?.stop()
+        binding.commentsRecyclerView.adapter = null
+    }
+
+    private fun projectContextViewClick() {
+        back()
+    }
+
+    private fun commentsLogin() {
+        val intent = Intent(this, LoginToutActivity::class.java)
+            .putExtra(IntentKey.LOGIN_REASON, LoginReason.COMMENT_FEED)
+        startActivityForResult(intent, ActivityRequestCodes.LOGIN_FLOW)
+    }
+
+    private fun commentButtonClicked() {
```

Comment:
nit picky detail, same for the inline functions for this method and for `dismissCommentDialog` and `projectContextViewClick`